integration/integration_test.go

```diff
@@ -206,42 +216,52 @@ func TestLayers(t *testing.T) {
 			kanikoImage := GetKanikoImage(config.imageRepo, dockerfile)
 			pullCmd := exec.Command("docker", "pull", kanikoImage)
 			RunCommand(pullCmd, t)
-			if err := checkLayers(dockerImage, kanikoImage, offset[dockerfile]); err != nil {
+			if err := checkLayers(t, dockerImage, kanikoImage, offset[dockerfile]); err != nil {
 				t.Error(err)
 				t.Fail()
 			}
 		})
 	}
 }
 
-func checkLayers(image1, image2 string, offset int) error {
-	lenImage1, err := numLayers(image1)
+func checkLayers(t *testing.T, image1, image2 string, offset int) error {
+	img1, err := getImageDetails(image1)
 	if err != nil {
-		return fmt.Errorf("Couldn't get number of layers for image1 (%s): %s", image1, err)
+		return fmt.Errorf("Couldn't get details from image reference for (%s): %s", image1, err)
 	}
-	lenImage2, err := numLayers(image2)
+
+	img2, err := getImageDetails(image2)
 	if err != nil {
-		return fmt.Errorf("Couldn't get number of layers for image2 (%s): %s", image2, err)
+		return fmt.Errorf("Couldn't get details from image reference for (%s): %s", image2, err)
 	}
-	actualOffset := int(math.Abs(float64(lenImage1 - lenImage2)))
+
+	actualOffset := int(math.Abs(float64(img1.numLayers - img1.numLayers)))
 	if actualOffset != offset {
-		return fmt.Errorf("incorrect offset between layers of %s and %s: expected %d but got %d", image1, image2, offset, actualOffset)
+		return fmt.Errorf("Difference in number of layers in each image is %d but should be %d. Image 1: %s, Image 2: %s", actualOffset, offset, img1, img2)
 	}
 	return nil
 }
 
-func numLayers(image string) (int, error) {
+func getImageDetails(image string) (*imageDetails, error) {
 	ref, err := name.ParseReference(image, name.WeakValidation)
 	if err != nil {
-		return 0, fmt.Errorf("Couldn't parse referance to image %s: %s", image, err)
+		return nil, fmt.Errorf("Couldn't parse referance to image %s: %s", image, err)
+	}
+	imgRef, err := daemon.Image(ref)
```

Comment:
Only one of these should be a daemon.Image, right? The other is remote?

Reply:
@dlorenc [the kaniko image is pulled](https://github.com/bobcatfish/kaniko/blob/f7b659d701b4788c8cf91d4a6e4221039857ff51/integration/integration_test.go#L217) right before this executes (maybe that should be made more clear somehow?)